cleverhans/attacks.py

```diff
@@ -1115,11 +1127,14 @@ def attack(self, x, y):
         import tensorflow as tf
         from cleverhans.utils_tf import clip_eta
 
-        eta = tf.random_uniform(tf.shape(x), -self.eps, self.eps)
-        eta = clip_eta(eta, self.ord, self.eps)
+        if self.rand_init:
+            eta = tf.random_uniform(tf.shape(x), -self.eps, self.eps)
+            eta = clip_eta(eta, self.ord, self.eps)
+        else:
+            eta = tf.zeros_like(x)
 
         for i in range(self.nb_iter):
-            x, eta = self.attack_single_step(x, eta, y)
+            eta = self.attack_single_step(x, eta, y)
 
         adv_x = x + eta
         if self.clip_min is not None and self.clip_max is not None:
```

Comment:
This file looks OK once `rand_init` has a docstring entry. In the future, it would be a good idea to send things like the new `rand_init` parameter as a separate PR. It's almost trivial to review on its own, but now that it's in this PR, it won't get merged until everything else in the PR is merged.

Reply:
Got it. I'll do so next time. Thanks.